### ISSUE TITLE
model.fit outputs an extra "Epoch 1/x" for each epoch

### ISSUE THREAD

**Issue by bersbersbers** [tensorflow/tensorflow#35815]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): see below
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10, Linux
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.1.0
- Python version: 3.7.6

**Describe the current behavior**
```
Epoch 1/2
Epoch 1/2
1/1 [==============================] - 0s 13ms/step - loss: 0.0000e+00
1/1 [==============================] - 0s 481ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 2/2
Epoch 1/2
1/1 [==============================] - 0s 1ms/step - loss: 0.0000e+00
1/1 [==============================] - 0s 5ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
```

**Describe the expected behavior**
"Epoch 1/2" should not be in each epoch.

**Code to reproduce the issue**
```
import tensorflow as tf

data = [[1]]
dataset = tf.data.Dataset.from_tensor_slices(data)
input_target = tf.data.Dataset.zip((dataset, dataset))
layer = tf.keras.layers.Input(shape=())
model = tf.keras.models.Model(inputs=layer, outputs=layer)
model.compile(optimizer="adam", loss="mse", experimental_run_tf_function=False)
model.fit(x=input_target, validation_data=input_target, epochs=2)
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by oanush** [comment]
@bersbersbers ,
Thats right, issue seems to be fixed in tf-nightly. [Colab Gist](https://colab.sandbox.google.com/gist/oanush/0c6b2dd594b3cfbbce82eb62ab01a2eb/35815.ipynb) for reference.

**Comment by oanush** [comment]
@bersbersbers ,
Any update on the issue ?Thanks!

**Comment by bersbersbers** [comment]
Not really, it's still and issue in 2.1, and I have no idea whether to expect that that this will be fixed in TF 2.1.x (or TF 2.2, for that matter).

**Comment by jvishnuvardhan** [comment]
@bersbersbers Yes, This is still an issue with `TF2.1`. It was update recently in `tf-nightly`. If you want to test, please install `tf-nightly` and test. Otherwise, there will be stable `TF2.2` version released in the near future. Thanks!

I am closing as it is resolved. Thanks!
